compiler/rustc_mir_build/src/build/scope.rs

```diff
@@ -1021,6 +1023,38 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         cached_drop
     }
 
+    /// This is similar to [diverge_cleanup] except its target is set to
```

Comment:
Afaict `diverge_cleanup` could now call this function and pass `self.scopes.topmost()` as the `target_scope` argument.